integration-tests/bats/sql-server.bats

```diff
@@ -1446,6 +1446,22 @@ databases:
     [ "$status" -eq 1 ]
 }
 
+@test "sql-server: sql-server sets permissions on sql-server.lock" {
```

Comment:
(minor) Might be nice to have a test somewhere that asserts the contents of the lock file; we probably have a bats test that checks for port that could maybe be extended easily. 

Might also be worth having a smoke test for the legacy lock file format, just to ensure we don't blow up if we see one. Seems like we'll need to keep that functionality around until 1.0 probably?

Reply:
This is actually pretty difficult to test legitimately. Since we don't have a previous version of the binary in the test, we can't produce a lock file in the old format (we could of course write one to disk manually).

We have several tests which validate that if we load a lock file which isn't understood that the binary blows it away and takes over. That's really only going to happen for us if they roll back their binary to a previous version. This all seems overboard for the amount of risk here.